channels/drive/client/drive_file.c

```diff
@@ -90,6 +90,34 @@ static BOOL drive_file_fix_path(WCHAR* path, size_t length)
 	return TRUE;
 }
 
+static BOOL contains_dotdot(const WCHAR* path)
+{
+	WCHAR dotdotbuffer[6] = { 0 };
+	const WCHAR* dotdot = InitializeConstWCharFromUtf8("..", dotdotbuffer, ARRAYSIZE(dotdotbuffer));
+	const WCHAR* tst = path;
+	const size_t len = _wcslen(path);
+
+	do
+	{
+		tst = _wcsstr(tst, dotdot);
+		if (!tst)
+			return FALSE;
+
+		/* Filter .. sequences in file or directory names */
+		if ((*(tst - 1) == L'/') || (*(tst - 1) == L'\\'))
```

Comment:
This will read invalid memory if `path` starts with `..` (not sure if this is a possible scenario though)

Reply:
no, we always start with an offset (the local prefix)